core/src/main/java/org/apache/calcite/plan/VisitorDataContext.java

```diff
@@ -172,6 +196,20 @@ private static RexNode removeCast(RexNode inputRef) {
     }
     return inputRef;
   }
-}
 
+  private static RexNode convertAndRemoveCast(RexBuilder builder, RexExecutor executor,
+                                     RexNode literal, RelDataType toType) {
```

Comment:
the name `literal` is a little bit misleading, because it may not be literal.

Reply:
Thanks. Updated.